p2p/transport/websocket/websocket_test.go

```diff
@@ -548,3 +548,71 @@ func TestResolveMultiaddr(t *testing.T) {
 		})
 	}
 }
+
+func TestListenerResusePort(t *testing.T) {
+	laddr := ma.StringCast("/ip4/127.0.0.1/tcp/5002/ws")
+
+	var opts []Option
+	opts = append(opts, EnableReuseport())
+	_, u := newUpgrader(t)
+	tpt, err := New(u, &network.NullResourceManager{}, opts...)
+	require.NoError(t, err)
+	c := make(chan int)
+
+	for i := 0; i < 2; i++ {
+		go func(index int, ch chan int) {
+			l, err := tpt.maListen(laddr)
```

Comment:
I don't understand this test. You're opening two listeners on the same local address. How do you now make sure that each of them receives exactly 4 incoming connections.

Reply:
That is a challenge, because whenever i ran the test..it is not gauranteed to distribute connections equally..I have noticed 1,3 or 2,2 happen. Hence i did not add an additional check for validating exactly equal connections in the test. Should've added comments in the test regarding this to make it more clear.